operator/src/main/java/org/keycloak/operator/KeycloakDeployment.java

```diff
@@ -0,0 +1,146 @@
+                .apps()
+                .deployments()
+                .inNamespace(keycloak.getMetadata().getNamespace())
+                .list()
+                .getItems()
+                .stream()
+                .filter((d) -> d.getMetadata().getName().equals(org.keycloak.operator.Constants.NAME))
+                .findFirst()
+                .orElse(null);
+//                .withName(Constants.NAME)
+//                .get();
+    }
+
+    public void createKeycloakDeployment(Keycloak keycloak) {
+        client
+            .apps()
+            .deployments()
+            .inNamespace(keycloak.getMetadata().getNamespace())
+            .create(newKeycloakDeployment(keycloak));
+    }
+
+    public Deployment newKeycloakDeployment(Keycloak keycloak) {
+        return new DeploymentBuilder()
+                .withNewMetadata()
+                    .withName(Constants.NAME)
+                    .withNamespace(keycloak.getMetadata().getNamespace())
+                    .addToLabels(Constants.MANAGED_BY_LABEL, Constants.MANAGED_BY_VALUE)
+                    .addNewOwnerReference()
+                        .withApiVersion(Constants.CRDS_VERSION)
+                        .withKind(keycloak.getKind())
+                        .withName(keycloak.getMetadata().getName())
+                        .withUid(keycloak.getMetadata().getUid())
+                    .endOwnerReference()
+                .endMetadata()
+
+                .withNewSpec()
+
+                .withNewTemplate()
+                .withNewMetadata()
+                .addToLabels(Constants.DEFAULT_LABELS)
+                .endMetadata()
+                .withNewSpec()
+
+                .addNewInitContainer()
+                .withName("init-container")
+                .withImage(Constants.DEFAULT_KEYCLOAK_INIT_IMAGE)
+                .endInitContainer()
+
+                .addNewContainer()
+                .withName(Constants.NAME)
+                .withImage(Constants.DEFAULT_KEYCLOAK_IMAGE)
+                .addNewPort()
+                .withContainerPort(8443)
+                .withProtocol("TCP")
+                .endPort()
+                .endContainer()
+                .endSpec()
+
+                .endTemplate()
+                .withReplicas(keycloak.getSpec().getInstances())
+                .withNewSelector()
+                .addToMatchLabels(Constants.DEFAULT_LABELS)
+                .endSelector()
+
+                .endSpec()
+                .build();
+    }
+
+    public KeycloakStatus getNextStatus(KeycloakSpec desired, KeycloakStatus prev, Deployment current) {
+
+        var isReady = (current != null &&
+                current.getStatus() != null &&
+                current.getStatus().getReadyReplicas() != null);
+
+        if (prev == null) {
+            var newStatus = new KeycloakStatus();
+            newStatus.setState(READY);
+            newStatus.setMessage("Keycloak deployment started");
+            return newStatus;
+        }
```

Comment:
I don't think this is correct. How can we assume the Deployment is ready just from the fact that the status was missing?

Reply:
it is not, but we need to use informers and improve everything here, in this PR we want to setup the project, the business logic is just dummy.

Reply:
@andreaTP Could you please mark this as resolved?